crates/sui-core/src/authority.rs

```diff
@@ -882,9 +878,7 @@ impl AuthorityState {
             &self.metrics.bytecode_verifier_metrics,
         )?;
 
-        if epoch_store.coin_deny_list_state_enabled() {
```

Comment:
seems like this check shouldn't be removed unless we are also deprecating/removing the associated protocol config option?

Reply:
This is in the handle_transaction path, so we don't need to worry about replay. Now that this is released to all networks, I don't think we still need the check.

Reply:
If we don't still need the check, instead of just removing it here, I think it should be fully deprecated so there's no confusion as to whether this setting does anything anymore. i.e. remove the `coin_deny_list_state_enabled` function (I think no callers are left?) and mark the protocol config field as deprecated (or delete that too if possible)